src/app/modules/main/wallet_section/activity/controller.nim

```diff
@@ -33,11 +33,12 @@ proc toRef*[T](obj: T): ref T =
 const FETCH_BATCH_COUNT_DEFAULT = 10
 const FETCH_RECIPIENTS_BATCH_COUNT_DEFAULT = 2000
 
-# TODO: implement passing of collectibles
 QtObject:
   type
     Controller* = ref object of QObject
       model: Model
+      modelLock: Lock
```

Comment:
Don't see many mutexes in status-desktop... I thought most usual flows (communication with status-go) used some kind of event queue in a single thread, where does the need for this come from?

Reply:
Hmm, thanks for raising this.
We were using signals and slots for the Nim async work reporting. Those are queued signals and serialized.
I don't remember if the messaging system uses a similar mechanism. I have to check, then this lock won't be needed.
If we deliver from the emitting thread, then it is given that the filter results and filter updates are coming from different goroutines.

Reply:
You were right. It seems the status go registered callback [here](https://github.com/status-im/status-desktop/blob/722a9022e292f15248c391915f1fe2ee69e07c3c/src/nim_status_client.nim#L76) calls [`SignalsManager.receiveSignal`](https://github.com/status-im/status-desktop/blob/17c4bfea1e2a68c36f911dbbac680cd055af019c/src/app/core/signals/signals_manager.nim#L55) which is a slot. The slot is invoked through [`dos_signal`](https://github.com/status-im/status-desktop/blob/722a9022e292f15248c391915f1fe2ee69e07c3c/vendor/DOtherSide/lib/src/DOtherSide.cpp#L1088) using [`QMetaObject::invokeMethod`](https://doc.qt.io/qt-5/qmetaobject.html#invokeMethod) using a [`Qt::QueuedConnection`](https://doc.qt.io/qt-5/qt.html#ConnectionType-enum)

I will remove the synchronization helpers.